src/OrchardCore.Modules/OrchardCore.Media/wwwroot/Scripts/media.js

```diff
@@ -605,6 +605,7 @@ function initializeMediaApplication(displayMediaApplication, mediaApplicationUrl
             })[0];
             media.mediaPath = newPath;
             media.name = newName;
+            media.url = media.url.substr(0, media.url.lastIndexOf(oldPath)).concat(newPath);
```

Comment:
Hmm, here i think we need `encodeURIComponent(somePath)`

That said we use a `gulpfile.js` that you can run e.g. under Visual Studio through the `Task Runer Explorer` extension. The gulp pipeline inputs come from the module `/Assets` folder and the generated files are outputed under its `wwwroot`. So here you rather need to update the `/Assets/js/MediaApp/app.js` file, and then run a gulp build / rebuild. You may have to run first `npm install` under the solution root, other devs will help you if you have any issue.

Reply:
Hi @jtkech , thanks for your kind guidance. I'm totally a freshman for these things. I made a try in my new commit. Hope it works this way.